### ISSUE TITLE
bug(android): Missing Predictive text on the Suggestion bar 

### ISSUE THREAD

**Issue by bharanidharanj** [keymanapp/keyman#7283]
**Describe the bug**
After randomly typing some letters on the text pane, I entered some valid text (eg., testing) and noticed that the predictive text is not working on the suggestion bar. 

**To Reproduce**
1. Install Keyman 16.0.61-alpha build. 
2. In Keyman, type 'Aqua' and we may notice that the related text appears on the suggestion bar. 
3. Enter some random letters on the text pane. 
4. Enter a spacebar key. 
5. Type a word 'testing'
6. No predictive text appear on the suggestion bar. 
7. Clear text
8. Type 'testing'. Here, I noticed that the predictive text is missing on the suggestion bar. 

**Expected behavior**
Predictive text should appear after clearing the text in the text pane. 

**Screenshots**

<img width="300" src="https://user-images.githubusercontent.com/19683143/189914125-71aeaf7a-b538-47d0-8140-0b70caf661d8.jpg">

---

**Keyman for Android:**
 - Device:               Redmi Note 8 Pro
 - OS:                  Android 11.0 
 - Keyman version: 16.0.61-alpha
 - Target Application: Keyman In-App

---

**Keyboard**
 - Keyboard name:    English - EuroLatin(SIL)
 - Keyboard version: 2.0
 - Language name:    English

**Comment by jahorton** [comment]
Yeah... I have a feeling this is a side-effect from #7037.  I'll try to look into it.

**Comment by jahorton** [comment]
I've changed the title to match some of my own personal observations; it seems that the  'enhanced timer' may not be cancelling itself when text doesn't lead to any suggestions that don't involve full (or near-full) context replacement, causing each keystroke to use maximum time and create the effect of lag.

**Comment by jahorton** [comment]
There may be interesting cross-effects to consider once #7241 lands as well.  I actually thought it might have been related, but it hasn't even landed yet.  That's near-certainly useful information... now to see where it leads.
